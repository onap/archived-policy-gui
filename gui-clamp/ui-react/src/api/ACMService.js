/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021-2022 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

export default class ACMService {

  static async getACMInstantiation() {

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instantiation');

    return response
  }

  static async createInstanceProperties(instancePropertiesTemplate) {

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instanceProperties', {
      method: 'POST',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(instancePropertiesTemplate),
    });

    return response
  }

  static async updateInstanceProperties(instanceName, version, instancePropertiesTemplate) {
    const params = {
      name: instanceName,
      version: version
    }

    const response = await fetch(window.location.pathname +
        'onap/policy/clamp/acm/v2/instanceProperties?' + (new URLSearchParams(params)), {
      method: 'PUT',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(instancePropertiesTemplate),
    });

    return response
  }

  static async deleteInstantiation(name, version) {
    const params = {
      name: name,
      version: version
    }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instanceProperties?' + (new URLSearchParams(params)), {
      method: 'DELETE',
      credentials: 'same-origin',
    });

    const data = await response;

    return data;
  }

  static async getInstanceOrderState(name, version) {
    const params = {
      name: name,
      version: version
    }
    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instantiationState'+ '?' + (new URLSearchParams(params)));

    const data = await response;

    return data;
  }

  static async changeInstanceOrderState(toscaObject) {
    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instantiation/command', {
      method: 'PUT',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(toscaObject)
    });

    return response
  }

  static async getToscaTemplate(name, version, instanceName) {
    const params = instanceName != null ?
      {
        name: name,
        version: version,
        instanceName: instanceName
      } :
      {
        name: name,
        version: version
      }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission/toscaservicetemplate' + '?' + (new URLSearchParams(params)));

    const data = await response;

    return data;
  }

  static async uploadToscaFile(toscaObject) {
    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission', {
      method: 'POST',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(toscaObject),
    });

    return response

  }

  static async deleteToscaTemplate(name, version) {
    const params = {
      name: name,
      version: version
    }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission' + '?' + (new URLSearchParams(params)),
      {
        method: 'DELETE'
      });

    const data = await response;

    return data;
  }

  static async getCommonOrInstanceProperties(name, version, instanceName, isCommon) {
    const params = instanceName != null ?
      {
        name: name,
        version: version,
        instanceName: instanceName,
        common: isCommon
      } :
      {
        name: name,
        version: version,
        common: isCommon
      }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission/getCommonOrInstanceProperties' + '?' + (new URLSearchParams(params)));

    return response;
  }

}
