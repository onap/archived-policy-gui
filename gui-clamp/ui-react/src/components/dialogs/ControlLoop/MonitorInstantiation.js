/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useEffect, useState } from "react";
import styled from "styled-components";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import InstantiationItem from "./InstantiationItem";
import ControlLoopService from "../../../api/ControlLoopService";
import InstantiationElements from "./InstantiationElements";
import { Alert } from "react-bootstrap";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`
const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const MonitorInstantiation = (props) => {
  const [show, setShow] = useState(true);
  const [controlLoopList, setControlLoopList] = useState([]);
  const [controlLoopInstantiationOk, setControlLoopInstantiationOk] = useState(true);
  const [controlLoopInstantiationError, setControlLoopInstantiationError] = useState({});

  useEffect(async () => {
    const controlLoopInstantiation = await ControlLoopService.getControlLoopInstantiation()
      .catch(error => error.message);

    const controlLoopInstantiationJson = await controlLoopInstantiation.json()
    console.log(controlLoopInstantiationJson)
    if (!controlLoopInstantiation.ok || controlLoopInstantiationJson['controlLoopList'].length === 0) {
      setControlLoopInstantiationOk(false)
      setControlLoopInstantiationError(controlLoopInstantiationJson)
    } else {
      setControlLoopList(controlLoopInstantiationJson[['controlLoopList']])
    }
  }, [])

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  return (
    <ModalStyled size="xl" show={ show } onHide={ handleClose } backdrop="static" keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Tosca Instantiation - Monitoring</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {
          controlLoopList.map((clList, index) => (
            <InstantiationItem title={ clList["name"] } orderedState={ clList["orderedState"] } index={ index } key={ index }>
              <InstantiationElements elements={ clList["elements"] } />
            </InstantiationItem>
          ))
        }
        <AlertStyled show={ !controlLoopInstantiationOk }
                     variant="danger">Can't get control loop instantiation info:<br/>{ JSON.stringify(controlLoopInstantiationError, null, 2) }</AlertStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" type="null" onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  )
}

export default MonitorInstantiation;
