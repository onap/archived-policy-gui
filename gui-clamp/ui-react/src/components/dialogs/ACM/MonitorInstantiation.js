/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useEffect, useState } from "react";
import styled from "styled-components";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import InstantiationItem from "./InstantiationItem";
import ACMService from "../../../api/ACMService";
import InstantiationElements from "./InstantiationElements";
import { Alert } from "react-bootstrap";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`
const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const MonitorInstantiation = (props) => {
  const [show, setShow] = useState(true);
  const [acmList, setAcmList] = useState([]);
  const [acmInstantiationOk, setAcmInstantiationOk] = useState(true);
  const [acmInstantiationError, setACMInstantiationError] = useState({});

  useEffect(async () => {
    const acmInstantiation = await ACMService.getACMInstantiation()
      .catch(error => error.message);

    const acmInstantiationJson = await acmInstantiation.json();

    if (!acmInstantiation.ok || acmInstantiationJson['automationCompositionList'].length === 0) {
      setAcmInstantiationOk(false)
      setACMInstantiationError(acmInstantiationJson)
    } else {
      setAcmList(acmInstantiationJson['automationCompositionList']);
    }
  }, [])

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  return (
    <ModalStyled size="xl" show={ show } onHide={ handleClose } backdrop="static" keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Tosca Instantiation - Monitoring</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {
          acmList.map((clList, index) => (
            <InstantiationItem title={ clList["name"] } orderedState={ clList["orderedState"] } index={ index } key={ index } >
              <InstantiationElements elements={ clList["elements"] } />
            </InstantiationItem>
          ))
        }
        <AlertStyled show={ !acmInstantiationOk }
                     variant="danger">Can't get acm instantiation info:<br/>{ JSON.stringify(acmInstantiationError, null, 2) }</AlertStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" type="null" onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  )
}

export default MonitorInstantiation;
