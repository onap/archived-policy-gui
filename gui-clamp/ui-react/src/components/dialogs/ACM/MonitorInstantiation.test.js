/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import { createMemoryHistory } from "history";
import MonitorInstantiation from "./MonitorInstantiation";
import ACMService from "../../../api/ACMService";
import clLoopList from "./testFiles/acmList.json";

const logSpy = jest.spyOn(console, 'log')
const history = createMemoryHistory();

describe('Verify MonitorInstantiation', () => {
  const flushPromises = () => new Promise(setImmediate);

  beforeEach(() => {
    logSpy.mockClear();
  });

  it("renders correctly", () => {
    const container = shallow(<MonitorInstantiation />);
    expect(toJson(container)).toMatchSnapshot();
  });

  it('should have a Button element', () => {
    const container = shallow(<MonitorInstantiation />);
    expect(container.find('Button').length).toEqual(1);
  });

  it('handleClose called when bottom button clicked', () => {
    const container = shallow(<MonitorInstantiation history={ history } />);
    const logSpy = jest.spyOn(console, 'log');

    act(() => {
      container.find('[variant="secondary"]').simulate('click');
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
  });

  it('handleClose called when top-right button clicked', () => {
    const container = shallow(<MonitorInstantiation history={ history } />);
    const logSpy = jest.spyOn(console, 'log');

    act(() => {
      container.find('[size="xl"]').get(0).props.onHide();
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
  });

  it('Check useEffect is being called', async () => {
    jest.resetAllMocks();
    jest.spyOn(ACMService, 'getACMInstantiation')
      .mockImplementationOnce(async () => {
        return Promise.resolve({
          ok: true,
          status: 200,
          text: () => "OK",
          json: () => {
            return Promise.resolve(clLoopList);
          }
        });
      });

    const component = mount(<MonitorInstantiation />);
    const useEffect = jest.spyOn(React, "useEffect");

    await act(async () => {
      await flushPromises()
      component.update();
      await expect(useEffect).toHaveBeenCalled();
    });
    component.unmount();
  });
});
