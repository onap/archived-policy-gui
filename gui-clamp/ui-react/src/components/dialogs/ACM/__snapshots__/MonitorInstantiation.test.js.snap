// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify MonitorInstantiation renders correctly 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Tosca Instantiation - Monitoring
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <Styled(Alert)
      show={false}
      variant="danger"
    >
      Can't get acm instantiation info:
      <br />
      {}
    </Styled(Alert)>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
